src/panels/calendar/ha-rrule.ts

```diff
@@ -0,0 +1,188 @@
+import { html, LitElement, TemplateResult } from "lit";
+import { state, customElement, property } from "lit/decorators";
+import { RRule, Frequency } from "rrule";
+import type { Options } from "rrule";
+import { fireEvent } from "../../common/dom/fire_event";
+import { stopPropagation } from "../../common/dom/stop_propagation";
+import "../../components/ha-textfield";
+import "../../components/ha-select";
+import "../../components/ha-selector/ha-selector-number";
+import { HomeAssistant } from "../../types";
+import { NumberSelector } from "../../data/selector";
+
+enum RepeatFrequency {
+  NONE = "none",
+  YEARLY = "yearly",
+  MONTHLY = "monthly",
+  WEEKLY = "weekly",
+  DAILY = "daily",
+}
+
+const convertFrequency = (freq: Frequency): RepeatFrequency | undefined => {
+  switch (freq) {
+    case Frequency.YEARLY:
+      return RepeatFrequency.YEARLY;
+    case Frequency.MONTHLY:
+      return RepeatFrequency.MONTHLY;
+    case Frequency.WEEKLY:
+      return RepeatFrequency.WEEKLY;
+    case Frequency.DAILY:
+      return RepeatFrequency.DAILY;
+    default:
+      return undefined;
+  }
+};
+
+const convertRepeatFrequency = (
+  freq: RepeatFrequency
+): Frequency | undefined => {
+  switch (freq) {
+    case RepeatFrequency.YEARLY:
+      return Frequency.YEARLY;
+    case RepeatFrequency.MONTHLY:
+      return Frequency.MONTHLY;
+    case RepeatFrequency.WEEKLY:
+      return Frequency.WEEKLY;
+    case RepeatFrequency.DAILY:
+      return Frequency.DAILY;
+    default:
+      return undefined;
+  }
+};
+
+@customElement("ha-rrule")
+export class HaRRule extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public mutable = true;
+
+  @property() public value?: string;
+
+  @state() private _freq?: RepeatFrequency;
+
+  @state() private _rrule?: Partial<Options>;
+
+  protected firstUpdated() {
```

Comment:
Don't use `firstUpdated` but use `willUpdate` and check if `value` was changed:
```ts
protected willUpdate(changedProps) {
  if (changedProps.has("value")) {
   ... calc `_freq` and `_rrule`
  }
}
```

Reply:
OK, i think where i was having trouble was also handling initialization. I've set an initial value for `_freq` instead which seems to address that.